internal/cli/serverless/sqluser/create.go

```diff
@@ -0,0 +1,302 @@
+	"tidbcloud-cli/internal/ui"
+	"tidbcloud-cli/internal/util"
+
+	iamApi "tidbcloud-cli/pkg/tidbcloud/v1beta1/iam/client/account"
+	iamModel "tidbcloud-cli/pkg/tidbcloud/v1beta1/iam/models"
+	serverlessApi "tidbcloud-cli/pkg/tidbcloud/v1beta1/serverless/client/serverless_service"
+
+	"github.com/charmbracelet/bubbles/textinput"
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/fatih/color"
+	"github.com/juju/errors"
+	"github.com/spf13/cobra"
+)
+
+type CreateOpts struct {
+	interactive bool
+}
+
+const (
+	WaitInterval = 5 * time.Second
+	WaitTimeout  = 2 * time.Minute
+)
+
+var createSQLUserField = map[string]int{
+	flag.UserName: 0,
+	flag.Password: 1,
+}
+
+func (c CreateOpts) NonInteractiveFlags() []string {
+	return []string{
+		flag.ClusterID,
+		flag.UserName,
+		flag.Password,
+		flag.UserRole,
+	}
+}
+
+func (c CreateOpts) RequiredFlags() []string {
+	return []string{
+		flag.ClusterID,
+		flag.UserName,
+		flag.Password,
+		flag.UserRole,
+	}
+}
+
+func CreateCmd(h *internal.Helper) *cobra.Command {
+	opts := CreateOpts{
+		interactive: true,
+	}
+
+	var CreateCmd = &cobra.Command{
+		Use:         "create",
+		Short:       "Create a SQL user",
+		Aliases:     []string{"c"},
+		Annotations: make(map[string]string),
+		Example: fmt.Sprintf(`  Create a SQL user in interactive mode:
+$ %[1]s serverless sql-user create
+
+Create a TiDB Serverless SQL user in non-interactive mode:
+$ %[1]s serverless sql-user create --user <user-name> --password <password> --role <role> --cluster-id <cluster-id>`,
+			config.CliName),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			flags := opts.NonInteractiveFlags()
+			for _, fn := range flags {
+				f := cmd.Flags().Lookup(fn)
+				if f != nil && f.Changed {
+					opts.interactive = false
+				}
+			}
+
+			// mark required flags in non-interactive mode
+			if !opts.interactive {
+				for _, fn := range opts.RequiredFlags() {
+					err := cmd.MarkFlagRequired(fn)
+					if err != nil {
+						return errors.Trace(err)
+					}
+				}
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			d, err := h.Client()
+			if err != nil {
+				return err
+			}
+
+			var clusterID string
+			var userName string
+			var password string
+			var userRole string
+			var userPrefix string
+			var customRoles []string
+			if opts.interactive {
+				cmd.Annotations[telemetry.InteractiveMode] = "true"
+				if !h.IOStreams.CanPrompt {
+					return errors.New("The terminal doesn't support interactive mode, please use non-interactive mode")
+				}
+
+				// interactive mode
+				project, err := cloud.GetSelectedProject(ctx, h.QueryPageSize, d)
+				if err != nil {
+					return err
+				}
+				projectID := project.ID
+
+				cluster, err := cloud.GetSelectedCluster(ctx, projectID, h.QueryPageSize, d)
+				if err != nil {
+					return err
+				}
+				clusterID = cluster.ID
+				userPrefix, err = GetUserPrefix(ctx, d, clusterID)
```

Comment:
why call `GetUserPrefix` here? we could already get userprefix from cluster object

Reply:
It's a mistake